Tests/anvil-csharp-tests.csproj

```diff
@@ -0,0 +1,23 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
```

Comment:
Should we match the main `anvil-csharp-core` csproj?
`netcoreapp3.1;netstandard2.1`

Reply:
Oh right - I tried that, but it yelled because NUnit uses cutting edge .NET features, so I left it as `net6.0` since this is just a unit test project, and that was the default when I created it.

Reply:
Seems like at least .NET Core 3.1 should be supported based on the release notes. 
https://docs.nunit.org/articles/nunit/release-notes/framework.html

Maybe it was just .NET Standard that was the issue?
https://docs.nunit.org/articles/nunit/getting-started/dotnet-core-and-dotnet-standard.html#why-cant-my-tests-target-net-standard

In any case, we should target the correct framework in case there are breaking, behavioural changes (unlikely) or performance changes.

The standard build actions should catch situations where a developer uses a feature in a newer version of .NET but it would be confusing so see the tests passing while the build action fails.

Reply:
Wrapping up an offline discussion:

The primary concern is that, IDEs can sometimes allow language features to be used and committed that aren't actually supported in a given project's target framework, after which the PR build step should fail, but we want to make sure the unit test step (when it exists) doesn't confusingly succeed.

This isn't a problem, because while the anvil-csharp-tests project targets .NET 6.0, it still compiles and references anvil-csharp-core separately, failing if any unsupported features are used. It is impossible for builds to fail and unit tests to pass.

---

In an attempt to generate a bit of empirical evidence, I ran `dotnet test` to confirm that each project is compiled independently:
```
  anvil-csharp-logging -> .\Logging\anvil-csharp-logging.dll
  anvil-csharp-core -> .\bin\Debug\netcoreapp3.1\anvil-csharp-core.dll
  anvil-csharp-tests -> .\Tests\bin\Debug\net6.0\anvil-csharp-tests.dll
```

...after which I added a `global using` (a .NET 6.0 / C# 10.0 feature) to a unit test file with no issue, but adding one to a random anvil-csharp-core file caused the build to fail as expected:
```
Core\AbstractAnvilBase.cs(2,1): error CS8400: Feature 'global using directive' is not available in C# 8.0. Please use language version 10.0 or greater. [anvil-csharp-core.csproj]
```

Reply:
Perfect, thanks for clearing that up for me and discussing. False alarm!